subprojects/distributions/src/integTest/groovy/org/gradle/DistributionIntegritySpec.groovy

```diff
@@ -191,6 +194,20 @@ class DistributionIntegritySpec extends DistributionIntegrationSpec {
         assert errors.empty
     }
 
+    @Issue(['https://github.com/gradle/gradle/issues/9990', 'https://github.com/gradle/gradle/issues/10038'])
+    def "validate dependency archives"() {
+        when:
+        def jars = collectJars(unpackDistribution())
+        def invalidArchives = jars.findAll {
+            def names = new ZipFile(it).entries()*.name
+            names.size() != names.toSet().size()
```

Comment:
Using `.toUnique()` would be more descriptive here.

Reply:
Done